### PR TITLE
hashindex: export max load factor to Python-space

### LINKED ISSUE #1580: make hashtable max load available to python
_hashindex.c: #define HASH_MAX_LOAD .75

We need that in Python code also, to compute a big enough hashtable size so it does not need to grow:

archive.py: capacity = int(len(self.repository) * 1.35 + 1) # > len * 1.0 / HASH_MAX_LOAD (see _hashindex.c)

So, make this constant available to python code via hashindex.pyx.

### PR DESCRIPTION
**Pull request by enkore**
Fixes #1580

### CODE REVIEW

File: borg/testsuite/hashindex.py [hunk 1]
```diff
@@ -266,13 +266,18 @@ def test_read_known_good(self):
...
+        idx[H(1)] = NSIndex.MAX_VALUE + 1, 0
     assert H(1) not in idx
-    idx[H(2)] = hashindex.MAX_VALUE, 0
+    idx[H(2)] = NSIndex.MAX_VALUE, 0
     assert H(2) in idx
+
+
+def test_max_load_factor():
+    assert NSIndex.MAX_LOAD_FACTOR < 1
+    assert ChunkIndex.MAX_LOAD_FACTOR < 1
```
Comment (ThomasWaldmann) [inline borg/testsuite/hashindex.py]: < 1.0 to make clear it is float?

**Comment by ThomasWaldmann** [comment]
you fix the one place also that started this discussion (see ticket)?

**Comment by enkore** [comment]
sure

File: borg/testsuite/hashindex.py (hunk 1 above)
Comment (ThomasWaldmann) [inline borg/testsuite/hashindex.py]: hmm, there is a comment near the load factor definition like "never go above 0.75" - maybe test for that here?

**Comment by ThomasWaldmann** [comment]
lgtm, except the 2 items i already pointed out.

**Comment by ThomasWaldmann** [comment]
ok, lgtm.

**Pull request merged**